### PR TITLE
Idms on

### PR DESCRIPTION
**Pull request by mpicchioni**
feat: Add IDMS DML support for ON autostatus clause #704

### CODE REVIEW

**Review by temanbrcom — COMMENTED** [review]
The last to commits may be squashed. 
We are researching what is going on with the build.

**Review by temanbrcom — APPROVED** [review]
🎸

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java [hunk 1]
```diff
@@ -51,19 +58,34 @@
...
+  @DisplayName("Parameterized - IDMS transfer tests")
   void test(String text) {
-    UseCaseEngine.runTest(text, ImmutableList.of(), ImmutableMap.of());
+    UseCaseEngine.runTest(
+        text,
+        ImmutableList.of(),
+        ImmutableMap.of(
+            "1",
+            new Diagnostic(
+                null, MESSAGE_1, DiagnosticSeverity.Error, SourceInfoLevels.ERROR.getText())));
```
Comment (grianbrcom) [inline server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java:89]: You expect the error only for one test, not for all. Please, extract this test and run it separately.
Right now UseCase Engine allows passing tests if the number of expected errors greater than actual. But we will fix it soon.
You can create a separate method:
```
@Test
void testWithError() {
    UseCaseEngine.runTest(
        BOILERPLATE + TRANSFER_NO_TERM,
        ImmutableList.of(),
        ImmutableMap.of(
            "1",
            new Diagnostic(
                null, MESSAGE_1, DiagnosticSeverity.Error, SourceInfoLevels.ERROR.getText())));
```

**Review by grianbrcom — CHANGES_REQUESTED** [review]
Please, rewrite tests.

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java (hunk 1 above)
Comment (mpicchioni) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java:89]: I have made the requested changes

Comment (temanbrcom) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java:89]: I don't see the changes, did you push them?

**Review by asatklichov — APPROVED** [review]
🦖

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java (hunk 1 above)
Comment (mpicchioni) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java:87]: Sorry about that, TestIdmsTransferStatement.java is updated now.

Comment (grianbrcom) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestIdmsTransferStatement.java:87]: Thanks)

**Review by grianbrcom — APPROVED** [review]
:bear:

**Pull request merged**